core/web/resolver/mutation.go

```diff
@@ -1030,7 +1038,7 @@ func (r *Resolver) UpdateChain(ctx context.Context, args struct {
 func (r *Resolver) DeleteChain(ctx context.Context, args struct {
 	ID graphql.ID
 }) (*DeleteChainPayloadResolver, error) {
-	if err := authenticateUser(ctx); err != nil {
+	if err := authenticateUserCanEdit(ctx); err != nil {
```

Comment:
One benefit of using an int based enum for UserRole is that you may be able to simlify this set of funcs in to one parameterized version that checks `<=`, e.g.
```suggestion
	if err := authenticateUser(ctx, sessions.UserRoleEdit); err != nil {
```

Reply:
That's a good idea, but maybe we can keep the current naive flexibility for now?